core/kernel/wait_queue.c

```diff
@@ -20,10 +20,10 @@ void wq_init(struct wait_queue *wq)
 	*wq = (struct wait_queue)WAIT_QUEUE_INITIALIZER;
 }
 
-static void do_notif(TEE_Result (*fn)(uint32_t), int id,
-		     const char *cmd_str __maybe_unused,
-		     const void *sync_obj __maybe_unused,
-		     const char *fname, int lineno __maybe_unused)
+static TEE_Result do_notif(bool sleep, int id, uint32_t timeout_ms,
```

Comment:
I think argument `sleep` should be renamed 'wait` since it is used to differentiate between a "notif-wait" request (with or without a timeout value that is set from `timeout_ms` value) and a "notif-send".

Reply:
Ok, I changed to use 'wait'.